rmf_site_editor/src/site_asset_io.rs

```diff
@@ -61,12 +64,13 @@ impl SiteAssetIo {
         asset_name: String,
     ) -> BoxedFuture<'a, Result<Vec<u8>, AssetIoError>> {
         Box::pin(async move {
-            let bytes = surf::get(remote_url.clone())
-                .recv_bytes()
-                .await
-                .map_err(|e| {
-                    AssetIoError::Io(io::Error::new(io::ErrorKind::Other, e.to_string()))
-                })?;
+            let mut req = surf::get(remote_url.clone());
+            if let Some(key) = &(*FUEL_API_KEY.lock().unwrap()) {
```

Comment:
It's not likely to be a problem, but we should still avoid using `.unwrap()` here. An `Err` result means the mutex was poisoned, which means whatever thread had the mutex locked panicked while holding it. The information inside might be fine or it might be mangled in some unpredictable way.

We can use [`into_inner`](https://doc.rust-lang.org/std/sync/struct.PoisonError.html#method.into_inner) on the `PoisonedError` to access the underlying data, and we can then reset it to `None` and print a `warn!` to the user telling them to input their API key again. This would be a way to gracefully recover from a mutex poisoning.

Reply:
I tried to address it but maybe not in the best way in https://github.com/open-rmf/rmf_site/pull/149/commits/91a8256eee8a8ff6ffffade21b518a20c7df31f1, lock poisoning wouldn't happen in the current use case (unless a string allocation failed, but in that case we would probably have many more issues to worry about!). For now it returns an error and fails to load the asset since the variable might be corrupted. I've seen the http library panicking when headers are given invalid values with messages like:

```
thread 'IO Task Pool (2)' panicked at 'called `Result::unwrap()` on an `Err` value: http::Error(InvalidHeaderValue)', /usr/local/google/home/yadunund/.cargo/registry/src/index.crates.io-6f17d22bba15001f/http-client-6.5.3/src/isahc.rs:58:42
```

So propagating a potentially corrupted string might cause other issues down the line. Anyway if you feel we should still consume it feel free to push directly!

Reply:
Your solution is definitely better than what I was proposing.

I did one last little iteration on it to clear out the key to `None` just in case it contains some corrupted data: [b13f1f0](https://github.com/open-rmf/rmf_site/pull/149/commits/b13f1f076425932de0abec6b933e3bf29dfd614c)